### PR TITLE
148097177 google sso registration

### PR DESCRIPTION
**Pull request by jsimpson-cc**
Please review. This covers several of the SSO stories.

### CODE REVIEW

**Comment by jsimpson-cc** [comment]
I will need to look at some of these spec test failures. I don't think this is all just Travis weirdness. Some of them it appears were depending on user sign up flows which have changed.

**Comment by jsimpson-cc** [comment]
Spec and cucumber tests should be fixed now. Please review.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -233,7 +233,19 @@ def check_for_select_portal_user_type
...
+        # TODO If there were some way render the underlying content of
+        # the omniauth_origin url beneath the modal signup popup, that would
+        # be nice.
+        #
+        # Also if this could be passed in, rather than stored in the
+        # session, that would. 
+        #
+        # Otherwise, this is not actually used. 
+        #
+        redirect_to portal_user_type_selector_path(omniauth_origin: session["omniauth.origin"])
```
Comment (scytacki) [inline app/controllers/application_controller.rb]: I don't see how this omniauth_origin parameter is use by the portal_user_type_selector controller or view.

File: docker-compose.yml [hunk 2]
```diff
@@ -11,18 +11,42 @@ services:
...
+
       PORTAL_FEATURES:
+
+      #
+      # OAuth keys
+      #
+      SCHOOLOGY_CONSUMER_KEY:
+      SCHOOLOGY_CONSUMER_SECRET:
+      GOOGLE_CLIENT_KEY:
+      GOOGLE_CLIENT_SECRET:
```
Comment (scytacki) [inline docker-compose.yml]: Could you add these to app_environment_variables.sample.rb as well.

File: app/assets/javascripts/components/signup/signup.js.coffee [hunk 3]
```diff
@@ -38,7 +38,7 @@ modulejs.define 'components/signup/signup',
       teacherData: null
 
     getDefaultProps: ->
-      signupText: "Sign Up for #{Portal.siteName}!"
+      signupText:   "Sign Up for #{Portal.siteName}!"
```
Comment (jsimpson-cc) [inline app/assets/javascripts/components/signup/signup.js.coffee]: I can remove all of the files under app/assets/javascripts/components/signup

File: app/assets/javascripts/components/signup/basic_data_form.js.coffee [hunk 4]
```diff
@@ -38,8 +38,10 @@ modulejs.define 'components/signup/basic_data_form',
       @setState password: model.password
 
     onBasicFormValid: ->
-      @setState canSubmit:  (   @refs.firstName.isValidAsync() &&
-                                @refs.lastName.isValidAsync()       )
+      {anonymous} = @props
+      @setState canSubmit:  (   !anonymous || (
+                                  @refs.firstName.isValidAsync() &&
+                                  @refs.lastName.isValidAsync()     )  )
```
Comment (jsimpson-cc) [inline app/assets/javascripts/components/signup/basic_data_form.js.coffee]: I can remove all of the files under app/assets/javascripts/components/signup

File: app/controllers/api/v1/teachers_controller.rb [hunk 5]
```diff
@@ -39,6 +39,23 @@ def create
...
+        current_user.save!
+        session['omniauth_email'] = nil
+      end 
+
+      if session[:omniauth_origin]
+        attributes["omniauth_origin"]   = session[:omniauth_origin]
+        session[:omniauth_origin]       = nil
+      end
+    
+      puts "Created teacher with attributes #{attributes}"
```
Comment (jsimpson-cc) [inline app/controllers/api/v1/teachers_controller.rb]: I will remove this.

File: app/controllers/authentications_controller.rb [hunk 6]
```diff
@@ -1,23 +1,63 @@
...
+      #
+      # For teachers we should set a valid email after creating
+      # the teacher user.
+      #
+      session[:omniauth_email]  = omniauth.info.email
+
+      #
+      # Return user to omniauth_origin after signup is complete.
+      #
+      session[:omniauth_origin] = origin
```
Comment (jsimpson-cc) [inline app/controllers/authentications_controller.rb]: I am not aware that this worked with schoology. 
This is an attempt to support "Log in from collection pages" from PT# 148097177
The intention is user is redirected back to the collections page they were viewing when they signed up.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jsimpson-cc) [inline app/controllers/application_controller.rb]: The props are passed down to the portal-pages signup component, which performs the redirect in the case of SSO signup completion. 

```
      //
      // For omniauth final step, simply redirect to omniauth_origin
      //
      if((studentData || teacherData) && this.props.omniauth) {
        console.log("INFO omniauth final step, redirect.", this.props);
        var data = this.state.studentData ? this.state.studentData : this.state.teacherData;
        window.location.href = data.omniauth_origin;
        return null;
      }

```

File: app/controllers/authentications_controller.rb (hunk 6 above)
Comment (scytacki) [inline app/controllers/authentications_controller.rb]: OK I get it now.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (scytacki) [inline app/controllers/application_controller.rb]: Ah I see it now.

**Comment by jsimpson-cc** [comment]
Now the cucumber tests are broken elsewhere. Looks like it is still related to the missing login box.

**Pull request merged**